pop_tools/regrid.py

```diff
@@ -0,0 +1,432 @@
+
+
+def gen_dest_grid(
+    method, dx=0.25, dy=0.25, lat_axis_bnds=None, lat_axis=None, lon_axis_bnds=None, lon_axis=None
+):
+    """Generates a destination grid to use with xesmf
+
+    Parameters
+    ----------
+    method: str
+       Method to use for generating the destination grid, options include 'regular_grid' or 'define_lat_aux'
+
+    dx: float default = 0.25
+       Longitudinal grid spacing to use in regular grid
+
+    dy: float default = 0.25
+       Latitudinal grid spacing to use in regular grid
+
+    lat_axis_bnds: list, array
+       Latitude axis bounds to use when defining the destination latitudes
+
+    lat_axis: list, array
+       Latitude axis to use when defining the destination latitudes
+
+    lon_axis_bounds: list, array
+       Longitude axis bounds to use when defining the destination longitudes
+
+    lon_axis: list, array
+       Longitude axis to use when defining the destination longitudes
+
+    Returns
+    -------
+    xarray.Dataset
+      An xarray.Dataset with the associated coordinates, formatted with lat, lon, lat_b, lon_b
+    """
+
+    reg_grid = xe.util.grid_global(dx, dy)
+
+    if method == 'regular_grid':
+        out_ds = reg_grid
+
+    elif method == 'lat_axis':
+        out_ds = xr.Dataset()
+
+        if lat_axis_bnds is not None:
+            lat_b = lat_axis_bnds
+
+            if lat_axis is not None:
+                lat = lat_axis
+
+            else:
+                lat = (lat_axis_bnds[1:] + lat_axis_bnds[:-1]) / 2
+
+        else:
+            raise TypeError('Missing defined latitude axis bounds')
+
+        if lon_axis_bnds is None:
+            lon_b = reg_grid.lon_b.values[0, :]
+
+        if lon_axis is None:
+            lon = reg_grid.lon.values[0, :]
+
+        else:
+            raise TypeError('Missing defined longitude axis bounds')
+
+        lons, lats = np.meshgrid(sorted(lon), lat)
+        lons_b, lats_b = np.meshgrid(sorted(lon_b), lat_b)
+
+        out_ds['lat'] = (('y', 'x'), lats)
+        out_ds['lon'] = (('y', 'x'), lons)
+        out_ds['lat_b'] = (('y_b', 'x_b'), lats_b)
+        out_ds['lon_b'] = (('y_b', 'x_b'), lons_b)
+
+    else:
+        raise AttributeError(
+            'Method not supported - only supporting regular lat/lon and manually defined lat/lon aux grid'
+        )
+
+    return out_ds.set_coords(['lat', 'lat_b', 'lon', 'lon_b'])
+
+
+def to_uniform_grid(obj, dst_grid, regrid_method='conservative', **kwargs):
+    """
+    Transform the POP C-Grid to a regular lat-lon grid, using similar grid spacing
+
+    Parameters
+    ----------
+    obj: `xarray.Dataset` or `xarray.DataArray`
+         Input dataset or dataarray to regrid
+    dx = float
+         Longitudinal grid spacing for output grid
+
+    dy = float
+         Latitudinal grid spacing for output grid
+
+    regrid_method = str
+         Regridding method for xESMF, default is conservative
+
+    **kwargs
```

Comment:
If I understand this correctly, these `kwargs` are meant to be passed to 

```python
def _regrid_dataset(da_in, dst_grid, regrid_method=None)
```

This function only has one keyword argument `regrid_method`. Could we just expose this argument in `to_uniform_grid` signature instead of using the catch-all `kwargs`?